StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -9,37 +9,25 @@ import UIKit
 
 /// For internal SDK use only.
 @objc(STP_Internal_DownloadManager)
-@_spi(STP) public class DownloadManager: NSObject, URLSessionDelegate {
+@_spi(STP) public class DownloadManager: NSObject {
     public typealias UpdateImageHandler = (UIImage) -> Void
 
-    private enum Error: Swift.Error {
-        case downloadSyncFailure
-        case downloadAsyncFailure
+    enum Error: Swift.Error {
+        case failedToMakeImageFromData
     }
 
     public static let sharedManager = DownloadManager()
 
-    let downloadQueue: DispatchQueue
-    let downloadOperationQueue: OperationQueue
-    let session: URLSession!
+    let imageCache = ImageCache()
```

Comment:
Why have this cache and URLCache?

Reply:
Good question! I incorrectly assumed this class previously had a in-memory cache and disk cache, with the `URLCache` being the disk cache. But turns out `URLCache` is also a in-memory cache so the `imageCache` was redundant. This should have been more obvious b/c we are passing in a `memoryCapacity` to the `URLCache`. 

> The URLCache class implements the caching of responses to URL load requests, by mapping [NSURLRequest](https://developer.apple.com/documentation/foundation/nsurlrequest) objects to [CachedURLResponse](https://developer.apple.com/documentation/foundation/cachedurlresponse) objects. It provides a composite in-memory and on-disk cache, and lets you manipulate the sizes of both the in-memory and on-disk portions.

https://developer.apple.com/documentation/foundation/urlcache

I removed `ImageCache` which also means we need to remove our cache specific tests since testing `URLCache` w/ OHHTTPStubs isn't supported, but I think that is ok given that we shouldn't really write tests for`URLCache`. I did add a test to ensure we are configuring the cache properly and attaching it to `URLSessionConfiguration`.